driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy

```diff
@@ -141,6 +141,29 @@ class ConnectionStringSpecification extends Specification {
                 .withWTimeout(5, MILLISECONDS).withJournal(true)
     }
 
+    @Unroll
```

Comment:
I wasn't confident in the test coverage from the spec so I added a few more tests here.